@here/harp-omv-datasource/lib/GeometryClipping.ts

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Vector2 } from "three";
+
+const tmpBA = new Vector2();
+const tmpQP = new Vector2();
+const tmpPA = new Vector2();
+const tmpA = new Vector2();
+const tmpB = new Vector2();
+
+/**
+ * Clip the given polygon using the Sutherland-Hodgman algorithm.
+ * @param polygon The polygon to clip
+ * @param clip The clip shape
+ * @return Clipped polygon
+ */
+export function clipPolygon(polygon: Vector2[], clip: Vector2[]): Vector2[] {
+    let outputList = polygon;
+    for (let e = 0; e < clip.length; ++e) {
+        const p = clip[e];
+        const q = clip[(e + 1) % clip.length];
+        const inputList = outputList;
+        outputList = [];
+        for (let i = 0; i < inputList.length; ++i) {
+            const currentPoint = inputList[i];
+            const prevPoint = inputList[(i + inputList.length - 1) % inputList.length];
+            if (inside(currentPoint, p, q)) {
+                if (!inside(prevPoint, p, q)) {
+                    outputList.push(computeIntersection(prevPoint, currentPoint, p, q));
+                }
+                outputList.push(currentPoint);
+            } else if (inside(prevPoint, p, q)) {
+                outputList.push(computeIntersection(prevPoint, currentPoint, p, q));
+            }
+        }
+    }
+    return outputList;
+}
+
+/**
+ * Clip given polyline to a clip shape.
```

Comment:
Not sure if we really need this generic implementation. Performance wise it would be better to not use an arbitrary clip shape but the tile border (i.e. 0 <= x,y <= 4096)